Code/GraphMol/QueryBond.cpp

```diff
@@ -67,6 +68,15 @@ void QueryBond::setBondDir(BondDir bD) {
 void QueryBond::expandQuery(QUERYBOND_QUERY *what,
                             Queries::CompositeQueryType how,
                             bool maintainOrder) {
+  bool thisIsNullQuery = dp_query->getDescription() == "BondNull";
+  bool otherIsNullQuery = what->getDescription() == "BondNull";
+
+  if (thisIsNullQuery || otherIsNullQuery) {
+    nullQueryCombine(dp_query, thisIsNullQuery, what, otherIsNullQuery, how);
+    delete what;
```

Comment:
It makes sense to do this `delete` (and I'm really glad you thought of it), but the comment in `QueryBond.h` should be modified to make it explicit that ownership of the `what` argument is being passed in `expandQuery()` and that calling code should no longer use the `what` pointer.
The same thing is true for `QueryAtom.h` above.

Reply:
Yeah, that makes sense. I had to guess that from the code, as the examples I found always got passed a Query::copy(), which returns a new pointer.